### PR TITLE
ROSAENG-65233: Add rosa-hyperfleet-zoa on-demand-e2e with dual-image injection

### PR DESCRIPTION
**Pull request by slopezz**
## Summary

[Jira: [ROSAENG-65233](https://redhat.atlassian.net/browse/ROSAENG-65233)](https://redhat.atlassian.net/browse/ROSAENG-65233)

Onboards `openshift-online/rosa-hyperfleet-zoa` to the shared hyperfleet ephemeral e2e workflow with dual-image build and PR auto-detection.

### CI operator changes

- Build **zoa-lambda** and **zoa-runner** images (replaces single `zoa-tools` image)
- **on-demand-e2e** job: `ROSA_REGIONAL_EXTRA_COMPONENTS` injects PR-built image tags and source repos into hyperfleet `config/defaults.yaml` so the ephemeral env runs the PR's Lambda/Runner

### Step registry

- **rosa-hyperfleet-e2e**: auto-detect `ZOA_REPO` / `ZOA_REF` from `rosa-hyperfleet-zoa` PRs (same pattern as `rosa-hyperfleet-api`)
- **rosa-hyperfleet-image-push**: wire `CI_ZOA_LAMBDA_IMAGE` and `CI_ZOA_RUNNER_IMAGE` deps for quay push

### Job timing

- **~40 min** ephemeral provision (RC + MC + pipelines)
- **~5 min** ZOA full e2e suite (RC+MC parallel, 79 specs each)
- No API / HCP / monitoring — zoa repo owns its validation path

## Companion PRs

- rosa-hyperfleet-zoa: https://github.com/openshift-online/rosa-hyperfleet-zoa/pull/77
- rosa-hyperfleet: https://github.com/openshift-online/rosa-hyperfleet/pull/789

> **Merge order**: zoa first→ hyperfleet→**this PR** . 
> Hyperfleet must not merge before zoa — presubmit smoke clones `zoa@main` and will fail until the e2e suite lands there. ,

## Test plan

- [x] `make update` (regenerate prow jobs) 
- [ ] `/test on-demand-e2e` on rosa-hyperfleet-zoa PR after this merges
- [ ] Verify image push to `quay.io/rrp-dev-ci/zoa-{lambda,runner}` with PR tag
- [ ] Verify `ROSA_REGIONAL_EXTRA_COMPONENTS` overrides land in provisioned env

[ROSAENG-65233]: https://redhat.atlassian.net/browse/ROSAENG-65233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- Onboards `openshift-online/rosa-hyperfleet-zoa` to the shared Hyperfleet ephemeral end-to-end workflow.
- Builds separate `zoa-lambda` and `zoa-runner` images for the workflow.
- Injects PR-built image tags and source repositories through `ROSA_REGIONAL_EXTRA_COMPONENTS`.
- Detects the ZOA repository and ref for `rosa-hyperfleet-zoa` pull requests.
- Adds both ZOA images to the Hyperfleet image-push dependencies.
- Regenerates the related Prow jobs with `make update`.

### CODE REVIEW

File: ci-operator/step-registry/rosa-hyperfleet/e2e/rosa-hyperfleet-e2e-commands.sh [hunk 1]
```diff
@@ -46,6 +46,19 @@ else
   export E2E_REPO="${DEFAULT_E2E_REPO}"
 fi
 
+# Same auto-detection for rosa-hyperfleet-zoa (ZOA e2e tests live in the zoa repo)
+if [[ -n "${ROSA_REGIONAL_ZOA_REF:-}" ]]; then
+  export ZOA_REF="${ROSA_REGIONAL_ZOA_REF}"
+  export ZOA_REPO="${ROSA_REGIONAL_ZOA_REPO:-https://github.com/openshift-online/rosa-hyperfleet-zoa.git}"
+elif [[ "${REPO_NAME:-}" == "rosa-hyperfleet-zoa" ]] && [[ -n "${PULL_NUMBER:-}" ]]; then
+  ZOA_REPO=$(curl -s "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}" | jq -r '.head.repo.clone_url')
```
[the remark this answers is not in the document]
Comment (slopezz) [reply ci-operator/step-registry/rosa-hyperfleet/e2e/rosa-hyperfleet-e2e-commands.sh:54]: Good call in general — this is a straight copy-paste of the existing rosa-hyperfleet-api PR auto-detect block a few lines above (same curl -s + jq -r '.head.repo.clone_url' pattern).

In openshift CI the GitHub API is reachable and fork PRs return a real clone_url; a bad response would fail at git clone anyway. I’d rather keep API and ZOA consistent here and harden both in a small follow-up if we want stricter validation.

File: ci-operator/step-registry/rosa-hyperfleet/e2e/rosa-hyperfleet-e2e-commands.sh [hunk 2]
```diff
@@ -46,6 +46,19 @@ else
   export E2E_REPO="${DEFAULT_E2E_REPO}"
 fi
 
+# Same auto-detection for rosa-hyperfleet-zoa (ZOA e2e tests live in the zoa repo)
+if [[ -n "${ROSA_REGIONAL_ZOA_REF:-}" ]]; then
+  export ZOA_REF="${ROSA_REGIONAL_ZOA_REF}"
+  export ZOA_REPO="${ROSA_REGIONAL_ZOA_REPO:-https://github.com/openshift-online/rosa-hyperfleet-zoa.git}"
+elif [[ "${REPO_NAME:-}" == "rosa-hyperfleet-zoa" ]] && [[ -n "${PULL_NUMBER:-}" ]]; then
```
[the remark this answers is not in the document]
Comment (slopezz) [reply ci-operator/step-registry/rosa-hyperfleet/e2e/rosa-hyperfleet-e2e-commands.sh:53]: Thanks for the review. The ref.yaml wording describes end-to-end behavior, not only what this step exports.

When we don’t export ZOA_REF / ZOA_REPO here (e.g. hyperfleet/api presubmit, nightly), rosa-hyperfleet/ci/e2e-tests.sh already applies the same defaults: ZOA_REF="${ZOA_REF:-main}" and ZOA_REPO="${ZOA_REPO:-https://github.com/openshift-online/rosa-hyperfleet-zoa.git}".

The ZOA block intentionally mirrors the existing API block above it — including the else branch only exporting the repo, not the ref (E2E_REF defaults in e2e-tests.sh the same way).

The only-ROSA_REGIONAL_ZOA_REPO-without-REF edge case is fair, but we don’t use that override today. Happy to tighten API + ZOA together in a follow-up if we start relying on independent repo/ref overrides.

**Comment by slopezz** [comment]
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-lint
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-test
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-verify
/pj-rehearse ack

**Comment by slopezz** [comment]
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-lint

**Comment by slopezz** [comment]
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-test

**Comment by slopezz** [comment]
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-verify

**Comment by typeid** [comment]
/lgtm
/approve
/hold 

Holding for merge queue / rehearsal ack once we can test it.

**Comment by slopezz** [comment]
/pj-rehearse pull-ci-openshift-online-rosa-hyperfleet-zoa-main-on-demand-e2e
